### DISCUSSION TITLE
Ctrl+Scroll to change object icon sizes

### DISCUSSION THREAD

**Discussion by headquarter8302** [spacedriveapp/spacedrive#1716] [Feature Request]
Parity with Windows Explorer's behaviour (and pretty much most explorers out there).

However, I've noticed that there's only two views available, which is the list view and the grid view, both of which are inflexible. So I guess this suggestion would be withheld for some time before the view modes have matured enough to have features added onto them

**Comment by utkubakir** [comment]
Note: Changing object sizes is not a thing on Macos so this might be a Windows only behavior

**Reply by mikekosulin** [reply]
Are you sure? You can adjust the sizes of objects in both the "list" and "icons" views.
Take a look at these two screens; it's necessary.

<img width="999" alt="finder app with larger list objects" src="https://github.com/spacedriveapp/spacedrive/assets/6089872/0cb9dd09-e1b8-480b-a0fc-11bd3f39f19a">


You have to try to use `cmd` + `+` / `-` in Finder:

<img width="959" alt="finder app with smaller list objects" src="https://github.com/spacedriveapp/spacedrive/assets/6089872/54eea23f-c3ae-407a-a04d-01add5dcafe6">

**Reply by utkubakir** [reply]
Didn't know this, we should definetetly implement it then. Scroll on windows and keys on macOS.

**Comment by paul-cine** [comment]
@headquarter8302 @utkubakir @mikekosulin 

List view in Finder using command + - is only 1 step up /  1 step down.

In Icon View it's zooming with command +/- based on the slider bottom right.  9 steps total. 

Better take a queue from Pathfinder, which can scale any view type from very small to huge. 
It autosaves every size you set for that specific view.  Which is really great. 

https://github.com/spacedriveapp/spacedrive/assets/34492742/469b1ef0-ffdd-4c81-8839-8c7e16790736

But Pathfinder has no keyboard mapping for this. 
I've added a screen capture to show

So what perhaps would be great for Space Drive in addition to mentioned above.

1. Scroll on windows and keys on macOS.
2. Font/icon size small/large unique for every view type. 
3.Command + - will zoom in/out relative to the current view/size 

sidenote_
As for the question mouse scroll question in general.  On Windows scrolling is often used for zooming. 
So a user that wants this on Mac would be better of to fix this on the system level. 
Which can be done pretty easily. Then every Mac app that uses command +- will just work whenever you scroll. 

He can take a look at **Karabiner, Better Touch Tool, or Smart Scroll**
https://karabiner-elements.pqrs.org/
https://marcmoini.com/sx_en.html

Programming route:
https://superuser.com/questions/369801/mac-os-x-zoom-in-application-with-mouse-wheel
https://apple.stackexchange.com/questions/98444/mouse-wheel-emulation-on-mac/372021#372021
https://gist.github.com/BalazsGyarmati/e199080a9e47733870889626609d34c5
